mlflow/sagemaker/__init__.py

```diff
@@ -1747,6 +1769,574 @@ def _does_model_exist(model_name, sage_client):
+        self.assumed_role_arn = None
+        self._get_values_from_target_uri()
+
+    def _get_values_from_target_uri(self):
+        parsed = urllib.parse.urlparse(self.target_uri)
+        values_str = parsed.path.strip("/")
+
+        if not parsed.scheme or not values_str:
+            return
+
+        separator_index = values_str.find("/")
+        if separator_index == -1:
+            # values_str would look like us-east-1
+            self.region_name = values_str
+        else:
+            # values_str could look like us-east-1/arn:aws:1234:role/assumed_role
+            self.region_name = values_str[:separator_index]
+            self.assumed_role_arn = values_str[separator_index + 1 :]
+
+            # if values_str contains multiple interior slashes such as
+            # us-east-1/////arn:aws:1234:role/assumed_role, remove
+            # the extra slashes that come before "arn"
+            self.assumed_role_arn = self.assumed_role_arn.strip("/")
+
+        if self.region_name.startswith("arn"):
+            raise MlflowException(
+                message=(
+                    "It looks like the target_uri contains an IAM role ARN without a region name.\n"
+                    "A region name must be provided when the target_uri contains a role ARN.\n"
+                    "In this case, the target_uri must follow the format: "
+                    "sagemaker:/region_name/assumed_role_arn.\n"
+                    f"The provided target_uri is: {self.target_uri}\n"
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+    def _default_deployment_config(self, create_mode=True):
+        config = dict(
+            assume_role_arn=self.assumed_role_arn,
+            execution_role_arn=None,
+            bucket=None,
+            image_url=None,
+            region_name=self.region_name,
+            archive=False,
+            instance_type=DEFAULT_SAGEMAKER_INSTANCE_TYPE,
+            instance_count=DEFAULT_SAGEMAKER_INSTANCE_COUNT,
+            vpc_config=None,
+            synchronous=True,
+            timeout_seconds=1200,
+        )
+
+        if create_mode:
+            config["mode"] = DEPLOYMENT_MODE_CREATE
+        else:
+            config["mode"] = DEPLOYMENT_MODE_REPLACE
+
+        return config
+
+    def _apply_custom_config(self, config, custom_config):
+        import json
+
+        int_fields = {"instance_count", "timeout_seconds"}
+        bool_fields = {"synchronous", "archive"}
+        dict_fields = {"vpc_config"}
+        for key, value in custom_config.items():
+            if key not in config:
+                continue
+
+            if key in int_fields and not isinstance(value, int):
+                value = int(value)
+            elif key in bool_fields and not isinstance(value, bool):
+                value = value == "True"
+            elif key in dict_fields and not isinstance(value, dict):
+                value = json.parse(value)
```

Comment:
There is some duplication in the conditionals. I attempted to DRY it up but the result seemed harder to follow. Please let me know if the duplication needs to be addressed.

Reply:
Love it! Looks great!